tests/monitors/elasticsearch_query/elasticsearch_query_test.py

```diff
@@ -0,0 +1,265 @@
+                        agent.fake_services,
+                        metric_name="elasticsearch_query.avg_cpu_utilization",
+                        dimensions={"index": "metrics", "metric_aggregation_type": "avg", "host_name": host},
+                    )
+                ), "Didn't get elasticsearch-query datapoints"
+
+
+@pytest.mark.parametrize("version", VERSIONS)
+def test_elasticsearch_query_terminal_bucket_aggs(version):
+    with run_service("elasticsearch/%s" % version) as es_container:
+        host = container_ip(es_container)
+        check_service_status(host)
+        write_data(host, version)
+
+        query = {"query": {"match_all": {}}, "aggs": {"host_name": {"terms": {"field": "host"}}}}
+
+        agent_config = AGENT_CONFIG_TEMPLATE.format(host=host, index="metrics", query=json.dumps(query))
+
+        with Agent.run(agent_config) as agent:
+            for host in HOSTS:
+                assert wait_for(
+                    p(
+                        has_datapoint,
+                        agent.fake_services,
+                        metric_name="elasticsearch_query.host_name.doc_count",
+                        dimensions={"index": "metrics", "bucket_aggregation_type": "terms", "host_name": host},
+                    )
+                ), "Didn't get elasticsearch-query datapoints"
+
+
+@pytest.mark.parametrize("version", VERSIONS)
+def test_elasticsearch_query_percentiles_aggs_with_terms_aggs(version):
+    with run_service("elasticsearch/%s" % version) as es_container:
+        host = container_ip(es_container)
+        check_service_status(host)
+        write_data(host, version)
+
+        query = {
+            "query": {"match_all": {}},
+            "aggs": {
+                "host_name": {
+                    "terms": {"field": "host"},
+                    "aggs": {"cpu_utilization_percentiles": {"percentiles": {"field": "cpu_utilization"}}},
+                }
+            },
+        }
+
+        agent_config = AGENT_CONFIG_TEMPLATE.format(host=host, index="metrics", query=json.dumps(query))
+
+        with Agent.run(agent_config) as agent:
+            for metric in PERCENTILE_METRICS:
+                for host in HOSTS:
+                    assert wait_for(
+                        p(
+                            has_datapoint,
+                            agent.fake_services,
+                            metric_name="elasticsearch_query.cpu_utilization_percentiles.%s" % metric,
+                            dimensions={
+                                "index": "metrics",
+                                "metric_aggregation_type": "percentiles",
+                                "host_name": host,
+                            },
+                        )
+                    ), "Didn't get elasticsearch-query datapoints"
+
+
+def write_data(host, version, num_docs=10):
+    """
+    Populates ES with mock data
+    """
+    es = Elasticsearch(hosts=[host])
+
+    doc_type = "doc"
```

Comment:
I think this needs to be `_doc` or just remove the `doc_type` shell object from the mapping definition.

Reply:
`_doc` is not a supported in 5.0.0 since it starts with an underscore. That's how I ended up with this. I'll try a different mapping to work around the issue.

Reply:
or actually this might be because the ``elasticsearch7.0.0` client is incompatible with ES 5.0.0. I'll try just using the REST API